### ISSUE TITLE
cloudflare_filter import issue - zone not set

### ISSUE THREAD

**Issue by SteveGoldthorpe-Work** [terraform-providers/terraform-provider-cloudflare#161]
- terraform import for cloudflare_filter resource forces a new resource on next apply as zone is not added up on import.
`    zone:        "" => "domain.name" (forces new resource)`

- terraform import for cloudflare_filter resource requires a zone_id rather than zone unlike the other resources.  We should be consistent.
`$ terraform import cloudflare_filter.default d41d8cd98f00b204e9800998ecf8427e/9e107d9d372bb6826bd81d3542a419d6`

provider.cloudflare 1.9

**Comment by jacobbednarz** [comment]
> terraform import for cloudflare_filter resource requires a zone_id rather than zone unlike the other resources.  We should be consistent.
>       `$ terraform import cloudflare_filter.default d41d8cd98f00b204e9800998ecf8427e/9e107d9d372bb6826bd81d3542a419d6`

@patryk did mention in an earlier comment that there was a push to go one or the other (zone vs zone ID) however I'm not sure where that is at. I don't think it's worth doing by itself and we should probably wait on that decision as there will need to be shuffling of the internals to achieve this. It's not a make or break thing at the moment and the import processes documented to reflect these key differences.

> terraform import for cloudflare_filter resource forces a new resource on next apply as zone is not added up on import.
>       `    zone:        "" => "domain.name" (forces new resource)`

This feels like a bug. Let me take a look at what we can do to handle setting a zone or zone ID and work out getting the other.

**Comment by patryk** [comment]
Re: import - yes, we are moving away from using zone names directly, so I'd like to new code use IDs instead. For existing things, we cannot do it without breaking change, so we will retrofit with IDs support. Eventually (sometime 1H2019) we'll do a 2.0 release where we remove all legacy constructs.

Re: empty field, yes, it's most definitely a bug.

**Comment by SteveGoldthorpe-Work** [comment]
I've just checked - ditto import for cloudflare_firewall_rule.
`zone:        "" => "domain.name" (forces new resource)`
I'm just wondering why zone_id is marked as computed and zone isn't.  Is it the wrong way around?  Or does zone_id get computed from zone in normal operation (just not import)?

**Comment by jacobbednarz** [comment]
@SteveGoldthorpe-Work I've just pushed up #162 which should aim to address this by keeping both in sync. Give it a whirl and let me know if you hit any snags.

**Comment by jacobbednarz** [comment]
@SteveGoldthorpe-Work I'd recommend we open a new issue for the issue above so that we can track it individually and not have it close when this one is resolved.

**Comment by SteveGoldthorpe-Work** [comment]
tested PR #162, zone is now set on import.
Moreover same code added to `cloudflare/resource_cloudflare_firewall_rule.go` fixes #165

**Comment by abohne** [comment]
I'm seeing similar behavior with a `cloudflare_zone` resource that has been imported.

**Comment by SteveGoldthorpe-Work** [comment]
I suspect the problem exists for anything importing using a zone id rather than a zone name.  I think they want a separate ticket for each resource for tracking purposes.
As I mentioned in another ticket, the workaround is to pull the state, edit the statefile to add the missing zone definition and push the modified state.
